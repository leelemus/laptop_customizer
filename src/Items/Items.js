import React, {Component} from 'react';

class Items extends Component {
    render () {
        return (
            <li key={this.props.index} className="feature__item">
                <div className={this.props.featureClass}
                  onClick={e => this.props.handleUpdate(this.props.featureName, this.props.item)}>
                    { this.props.item.name }
                    ({ new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                      .format(this.props.item.cost) })
                </div>
            </li>
        )
    };
}


export default Items;
