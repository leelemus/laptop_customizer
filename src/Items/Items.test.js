import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Items from './Items';


describe(`Items Component`, () => {

    const key = 'name'
    const index = 1
    const item = {
        name: 'Toyota Corolla 1.5v',
        cost: 1150.98
      };
    const featureClass = 'feature__option '
    
    it('renders Items with data', () => {
        const wrapper = 
            shallow(<Items 
                        key = {index}
                        index = {index}
                        item = {item}
                        featureName = {key}
                        featureClass = {featureClass}
                    />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
