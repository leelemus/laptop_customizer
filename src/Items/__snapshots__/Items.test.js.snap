// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Component renders Items with data 1`] = `
<li
  className="feature__item"
  key="1"
>
  <div
    className="feature__option "
    onClick={[Function]}
  >
    Toyota Corolla 1.5v
    (
    $1,150.98
    )
  </div>
</li>
`;
